### PR TITLE
ddl: check duplicated cols when execute alter table add index

### LINKED ISSUE #2091: Find duplicate column name when adding an index
1. What did you expect to see?
mysql> create table t (a int, b int);
Query OK, 0 rows affected (0.01 sec)

mysql> alter table t add index c (b,a,b);
ERROR 1060 (42S21): Duplicate column name 'b'

2. What did you see instead?
mysql> create table t (a int, b int);
Query OK, 0 rows affected (0.00 sec)

mysql>  alter table t add index c (b,a,b);
Query OK, 0 rows affected (0.01 sec)

### PR DESCRIPTION
**Pull request by XuHuaiyu**
fix #2091 
This commit checks whether exists duplicated column names when execute sql like
`alter table t add index c (b, a, B)`.

PTAL @shenli @coocood @hanfei1991 @zimulala @tiancaiamao @winoros

### CODE REVIEW

File: ddl/ddl.go [hunk 1]
```diff
@@ -1429,6 +1433,20 @@ func findCol(cols []*model.ColumnInfo, name string) *model.ColumnInfo {
...
 }
 
+// checkCreateIndexGrammar checks if index exists duplicated columns.
+func checkIndexGrammer(indexColNames []*ast.IndexColName) error {
+	for i := 0; i < len(indexColNames); i++ {
+		name1 := indexColNames[i].Column.Name
+		for j := i + 1; j < len(indexColNames); j++ {
+			name2 := indexColNames[j].Column.Name
+			if name1.L == name2.L {
+				return errors.Errorf("Duplicate column name '%s'", name2.O)
```
Comment (hanfei1991) [inline ddl/ddl.go]: return infoschema.ErrColumnExists.GenByArgs(name2)

**Comment by XuHuaiyu** [comment]
@hanfei1991 PTAL

File: ddl/ddl.go [hunk 2]
```diff
@@ -1429,6 +1433,20 @@ func findCol(cols []*model.ColumnInfo, name string) *model.ColumnInfo {
 	return nil
 }
 
+// checkCreateIndexGrammar checks if index exists duplicated columns.
+func checkIndexGrammer(indexColNames []*ast.IndexColName) error {
```
Comment (shenli) [inline ddl/ddl.go]: checkDuplicateColumnName is a better name.

**Comment by XuHuaiyu** [comment]
All test cases in `tidb-test` can be passed.

PTAL @shenli @hanfei1991

**Comment by hanfei1991** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**